### PR TITLE
Collapse All Groups in Collection View fixes #432

### LINKED ISSUE #432: Initialize all collapsed
Is there current functionality to initialize all groups collapsed?

### PR DESCRIPTION
**Pull request by jeffreyjackson**

### CODE REVIEW

**Comment by jeffreyjackson** [comment]
cc @brad36

**Comment by 3lvis** [comment]
Hi Jeffrey! Thanks a lot for your contribution, this looks very useful. :tada:

File: Source/FORMDataSource.m [hunk 1]
```diff
@@ -394,6 +394,24 @@ - (BOOL)isEnabled {
     return !_disabled;
 }
 
+- (void)collapseAllGroupsForCollectionView:(UICollectionView *)collectionView {
+    NSMutableArray *indexPaths = [NSMutableArray new];
+    [self.collapsedGroups removeAllObjects];
+    
+    int groupCount = 0;
+    for (FORMGroup *formGroup in self.formData.groups) {
```
Comment (3lvis) [inline Source/FORMDataSource.m]: could you use 

`[self.formData.groups enumerate....]` ? It would give you the index instead of doing `groupCount++`

**Comment by 3lvis** [comment]
Could you add a unit test for this method? For example calling the method then checking the count of elements inside `self.collapsedGroups`.

**Comment by jeffreyjackson** [comment]
Hey @3lvis - what do you think now?

File: Tests/Tests/FORMDataSourceTests.m [hunk 2]
```diff
@@ -881,4 +881,19 @@ - (void)testDynamicFormulas {
...
+    
+    FORMDataSource *dataSource = [[FORMDataSource alloc] initWithJSON:JSON
+                                                       collectionView:nil
+                                                               layout:nil
+                                                               values:nil
+                                                             disabled:YES];
+    
+    [dataSource collapseAllGroupsForCollectionView:nil];
+    
+    XCTAssertTrue([dataSource.collapsedGroups count] == [dataSource.groups count]);
```
Comment (3lvis) [inline Tests/Tests/FORMDataSourceTests.m]:
```objc
XCTAssertEqual([dataSource.collapsedGroups count], [dataSource.groups count]);
```

Comment (3lvis) [inline Tests/Tests/FORMDataSourceTests.m]: I'll do it in a separate PR :)

**Pull request merged**